pkg/app/piped/platformprovider/ecs/client.go

```diff
@@ -275,8 +275,9 @@ func (c *client) GetServiceTaskSets(ctx context.Context, service types.Service)
 		activeTaskSetArns = append(activeTaskSetArns, *svc.TaskSets[i].TaskSetArn)
 	}
 
+	var taskSets []*types.TaskSet
 	if len(activeTaskSetArns) == 0 {
-		return nil, fmt.Errorf("failed to get task sets of service %s: services empty", *service.ServiceName)
+		return taskSets, nil
```

Comment:
imo: just return `[]*types.TaskSet{}` and define `taskSets` variable to minify its scope. 👀

Reply:
And it should be better to add comment for the reason.

Reply:
Okey, I don't have strong opinion on this 👍 Let's me adopt your suggestion

Reply:
@ffjlabo addressed by https://github.com/pipe-cd/pipecd/pull/4614/commits/8c6060cb8fe70574509af60da2ea5e2ebed40648